README.md

```diff
@@ -619,7 +621,283 @@ const client = new Ably.Realtime({
+```
+
+The authentication token must include corresponding capabilities for the client to interact with Live Objects.
+
+#### Obtaining the Root Object
+
+The root object represents the top-level entry point for Live Objects state within a channel. It acts as a container for all nested data structures and must be retrieved in order to interact with the existing state.
+
+```typescript
+const root = await liveObjects.getRoot();
+```
+
+The root object is always a `LiveMap` instance and serves as the starting point for storing and organizing Live Objects state.
+
+#### Live Object Types
+
+Live Objects currently support two primary data structures:
+
+`LiveMap` - A key/value map data structure, similar to a JavaScript `Map`, where all changes are synchronized across clients in realtime. It allows you to store primitive values and other Live Objects, enabling a composable state.
+
+You can use `LiveMap` as follows:
+
+```typescript
+// root object is a LiveMap
+const root = await liveObjects.getRoot();
+
+// set a key on map
+await root.set('name', 'Alice');
+// get a value for a key
+console.log(root.get('name')); // output: 'Alice'
+
+// remove a key
+await root.remove('name');
+console.log(root.get('name')); // output: undefined
+
+// different data types are supported
+await root.set('foo', 'str');
+await root.set('bar', 1);
+await root.set('baz', true);
+await root.set('qux', new Uint8Array([21, 31]));
+// as well as other live objects
+const counter = await liveObjects.createCounter();
+await root.get('quux', counter);
+console.log(root.size()); // output: 5
+```
+
+`LiveCounter` - A counter that can be incremented or decremented and is synchronized across clients in realtime
+
+You can use `LiveCounter` as follows:
+
+```typescript
+const counter = await liveObjects.createCounter();
+
+// get current value of a counter
+console.log(counter.value()); // output: 0
+
+// increments and decrements
+await counter.increment(5);
+console.log(counter.value()); // output: 5
+
+await counter.decrement(2);
+console.log(counter.value()); // output: 3
+```
+
+#### Subscribing to Updates
+
+Subscribing to updates on Live Objects allows you to receive changes made by other clients in realtime. Since multiple clients may modify the same Live Objects state, subscribing ensures that your application reacts to external updates as soon as they are received.
+
+Additionally, mutation methods such as `LiveMap.set`, `LiveCounter.increment`, and `LiveCounter.decrement` do not directly edit the current state of the object locally. Instead, they send the intended operation to the Ably system, and the change is applied to the local object only when the corresponding realtime operation is echoed back to the client. This means that the state you retrieve immediately after a mutation may not reflect the latest updates yet.
+
+To ensure you react to state changes at the correct moment, you can subscribe to updates on all Live Objects as follows:
```

Comment:
```suggestion
You can subscribe to updates on all Live Objects using subscription listeners as follows:
```

I've added "subscription listeners" here because later way say you can de-register "subscription listeners" without mentioning them before. Obviously you can see from the code, but adding clarity here.